include/nighthawk/common/factories.h

```diff
@@ -33,12 +33,21 @@ class StatisticFactory {
   virtual StatisticPtr create() const PURE;
 };
 
+class RequestSourceConstructorInterface {
```

Comment:
I am a little concerned about the complexity of these changes.  Some of this might be naming but differentiating between a constructor and a factory here is pretty confusing, as both of them generally accomplish the same thing.

The more standard way to use a factory interface would be more like what Adaptive Load does, defining a factory that takes in something configurable like a proto.  I don't think it's necessary to ahve two separate methods here, createStaticRequestSource and createRemoteRequestSource, as those sound like implementation details.  And without that, I don't know what the ConstructorInterface here brings.

I'd like to schedule us a meeting to talk about this, if that's okay with you.  I'm sure there are things in the test code you're referring to that I'm not seeing for why this makes things less complex, and I'd like to get on the same page as you.